### DISCUSSION TITLE
Improvement on the label "Delete Resource" to take in consideration resource->label() or ->singularLabel()

### DISCUSSION THREAD

**Discussion by brunocfalcao** [laravel/nova-issues#5403] [Ideas & Feature Requests] [answered]
Hi team,

Quick improvement proposal, since we have the Resource ->label() that already improves the overall CRUD buttons, I think the "delete resource" should also take that in consideration.

![image](https://user-images.githubusercontent.com/34269950/224840955-8b0a2323-4b29-4b85-bdb6-27f4a2e262d8.png)

Instead of delete Resource, if I have defined, for e.g.: ->label() or singularLabel() returning "client" it could say "delete client"

Let me know your thoughts 👍 

Cheers
Bruno

**Comment by HugoHeneault** [comment]
This would be great! Related: https://github.com/laravel/nova-issues/discussions/6137

**Reply by ekazda** [reply]
This should all work in a similar way as how systems like WordPress handle [Custom Post Type creation](https://developer.wordpress.org/reference/functions/register_post_type/). Imaging if you couldn't rename various outputs inside a WordPress site?! Clients would go crazy!

**Comment by pakogn** [comment]
@brunocfalcao @ekazda 

Do you know if this was applied at some time?

thanks!

**Reply by justindantzer** [reply]
@pakogn 

Just had a client request for this, however, it does not seem to be implemented.

While it would be ideal that this be baked in, having acquired a link to [this package](https://github.com/ianrobertsFF/nova-override-example) that demos replacing Nova components, I found it pretty simple to customize the `DetailActionDropdown` to display the resource label, as it already has references to the `resourceInformation.singularLabel` value.

<img width="177" alt="Screenshot 2024-10-22 at 3 56 10 PM" src="https://github.com/user-attachments/assets/011b0bb4-32c5-422e-bc8d-e8955b4ddf98">


However, if you are wanting to account for all the relevant components that reference the generic "Resource" label, you'll need to modify several more. (at first glance, theres the DeleteResourceModal, RestoreResourceModal, and the components that reference those modals)

**Reply by pakogn** [reply]
Hey,

yeah! I ended taking that as a reference, it worked like a charm. Thanks!

**Comment by crynobone** [comment, marked as answer]
Will be available in 5.2.0

**Reply by crynobone** [reply]
Released
